### PR TITLE
Make CONFIGURE CURRENT DATABASE block until configuration is actually set

### LINKED ISSUE #7330: Database configure not always properly applying immediately on current connection
`test_server_proto_configure_error` is flaking occasionally: https://github.com/edgedb/edgedb/actions/runs/9007197736/job/24746488573

I would think it could *also* be cache related, but I think I saw the reverse happen too: the error having *not* been cleared during teardown.

I'm probably going to put a flake workaround

### PR DESCRIPTION
**Pull request by msullivan**
... on the local instance, at least. Multi-frontend will still be racy.

Fixes #7330.

### CODE REVIEW

**Comment by msullivan** [comment]
I think the background introspections were the cause of the flakes disabled by https://github.com/edgedb/edgedb/pull/7859.
I think they caused the schema to go backwards temporarily.

File: edb/server/tenant.py [hunk 1]
```diff
@@ -1656,29 +1657,18 @@ async def task():
...
-                            b'SELECT edgedb._clear_query_cache()')
-                        self._dbindex.get_db(dbname).clear_query_cache()
-            except Exception:
-                metrics.background_errors.inc(
-                    1.0, self._instance_name, "on_local_database_config_change"
-                )
-                raise
-
-        self.create_task(task(), interruptable=True)
+        # TODO: Can we just do config?
```
Comment (fantix) [inline edb/server/tenant.py:1663]: I think we do need the user schema for extension config specs.

File: edb/server/tenant.py [hunk 2]
```diff
@@ -1656,29 +1657,18 @@ async def task():
...
+        if await self.introspect_db(dbname):
```
Comment (fantix) [inline edb/server/tenant.py:1664]: self-note: `introspect_db()` has a lock inside, so concurrent database config changes from different frontends will be properly (?) serialized.

**Review by fantix — APPROVED** [review]
Yeah I think this is good, maybe @1st1 wanna have a look too?

**Review by fantix — APPROVED** [review]

**Pull request merged**